ansible/roles/operator-pipeline/templates/openshift/tasks/parse-repo-changes.yml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: parse-repo-changes
+spec:
+  params:
+    - name: pipeline_image
+    - name: head_commit
+      description: Commit ID of the head of the PR
+    - name: base_commit
+      description: Commit ID of the base of the PR
+  results:
+    - name: extra_files
+      description: >
+          Comma separated list of affected files not belonging to any operator
+    - name: added_operators
+      description: >
+          Comma separated list of new operators added
+    - name: modified_operators
+      description: >
+          Comma separated list of existing operators that have been modified
+    - name: deleted_operators
+      description: >
+          Comma separated list of operators that have been deleted
+    - name: added_bundles
+      description: >
+          Comma separated list of new bundles that have been added.
+          Bundle names are in the format "operator_name/bundle_version".
+    - name: modified_bundles
+      description: >
+          Comma separated list of existing bundles that have been modified
+          Bundle names are in the format "operator_name/bundle_version".
+    - name: deleted_bundles
+      description: >
+          Comma separated list of bundles that have been deleted
+          Bundle names are in the format "operator_name/bundle_version".
+  workspaces:
+    - name: repo
+      description: Local repo clone
+  steps:
+    - name: parse-repo-changes
+      image: "$(params.pipeline_image)"
+      script: |
+        #! /usr/bin/env bash
+        set -xe
+
+        cd "$(workspaces.repo.path)"
+        git config --global --add safe.directory "$(workspaces.repo.path)"
+
+        detect-changed-operators \
+          --repo-path="$(workspaces.repo.path)" \
+          --head-commit="$(params.head_commit)" \
+          --base-commit="$(params.base_commit)" \
+          --output-file /tmp/changes.json       \
+          --verbose
+
+        jq -r '.extra_files|join(",")' </tmp/changes.json >$(results.extra_files.path)
+        jq -r '.added_operators|join(",")' </tmp/changes.json >$(results.added_operators.path)
+        jq -r '.modified_operators|join(",")' </tmp/changes.json >$(results.modified_operators.path)
+        jq -r '.deleted_operators|join(",")' </tmp/changes.json >$(results.deleted_operators.path)
+        jq -r '.added_bundles|join(",")' </tmp/changes.json >$(results.added_bundles.path)
+        jq -r '.modified_bundles|join(",")' </tmp/changes.json >$(results.modified_bundles.path)
+        jq -r '.deleted_bundles|join(",")' </tmp/changes.json >$(results.deleted_bundles.path)
+
+        function fail() {
+            echo "ERROR: $*"
+            exit 1
+        }
+
+        jq -e '.extra_files|length==0' </tmp/changes.json \
+            || fail "The PR affects non-operator files"
+        jq -e '.affected_operators|length==1' </tmp/changes.json \
```

Comment:
Do you want to support adding multiple operator bundles in the single pipeline? How are we going to handle it in the rest of the pipeline?

Reply:
Since this is one of the features supported in the current implementation, I'd like to keep it if possible. I think we should be able to handle it with the Matrix construct in tekton. If it will prove to be too difficult, we can always add the relevant check here and abort the pipeline upfront.

Reply:
I am a bit worried about this feature because it will complicate all the remaining tasks in the Tekton pipeline and it also makes prow jobs more complicated than needed which increases the complexity that we will need to provide when dealing with issues.

If we want to provide the same user experience as for the ISV pipeline we should keep the logic the same (1PR == 1 operator bundle).

Reply:
Fair enough. I've just added the validation on the number of added bundles.